solutions/silver/usaco-669.mdx

```diff
@@ -419,3 +423,131 @@ public class Moocast {
 
 </JavaSection>
 </LanguageSection>
+
+### Implementation (Minimum Spanning Tree)
+
+**Time Complexity:** $\mathcal{O}(N^2)$
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+int d2(pair<int, int> &a, pair<int, int> &b) {
+	int dx = a.first - b.first;
+	int dy = a.second - b.second;
+	return dx * dx + dy * dy;
+}
+
+int main() {
+	freopen("moocast.in", "r", stdin);
+	freopen("moocast.out", "w", stdout);
+
+	int N;
+	cin >> N;
+	vector<pair<int, int>> cows(N);
+	for (auto &[x, y] : cows) cin >> x >> y;
+
+	vector<int> dist(N, INT_MAX - 1);
```

Comment:
why the -1

Reply:
it seems at the bottom you set dist[i] = INT_MAX so you don't go through i in the for loop
why not just check if j == i instead of dist[j] == INT_MAX?

Reply:
alr my bad
in that case add a comment
"INT_MAX - 1 indicates never been explored, INT_MAX means it has"